File: PackageInstaller.py
class PackageInstaller():

    def __init__(self):
        pass
    

    def install(self,dependencies):
        n = len(dependencies)
        
        indegree={}
        graph={}
        for node in dependencies:
            if node not in indegree:
                indegree[node]=0
            indegree[node]+=len(dependencies[node])
            for child in dependencies[node]:
                if child not in graph:
                    graph[child]=[]
                graph[child].append(node)
        x =self.topological_sort(graph,indegree)
        return x

        

    def topological_sort(self,graph, indegree):
        if not graph:
            return []
        order =[]
        queue=[]
        for node in indegree:
            if indegree[node]==0:
                queue.append(node)

        while queue:
            node =  queue.pop(0)
            order.append(node)
            #print(order)
            if node in graph:
                for child in graph[node]:
                    indegree[child]-=1
                    if indegree[child]==0:
                        queue.append(child)

        return order

 
if __name__ == '__main__':
    dependencies = {
        0:[1],
        1:[0]
    }

    packageInstaller = PackageInstaller()
    print(packageInstaller.install(dependencies))

File: PackageInstaller/src/PackageInstaller.java
import javax.print.attribute.HashDocAttributeSet;
import java.util.*;

enum NodeState{
    New,
    Installed,
    Deleted

}


class Node{
    String name;
    NodeState state;

    public Node(String name){
        this.name = name;
        state = NodeState.New;
    }

}

public class PackageInstaller {
      /*
      * [(1,0)]
      * [(1,0),(0,1)]
      *
      *
      *
      *
      * * * * * * */


    public boolean install(Map<Integer, List<Integer>> dependencies){
        try {
                    HashMap<Integer,Integer> graph =  new HashMap<>();
                    for(Integer node : dependencies.keySet()){

                    }


        }
        catch(RuntimeException ex){
            System.out.println("Encountered an exception"+ex.toString());
            return false;
        }
        return true;




    }

    public List<Integer> installOrder(){

        return null;
    }


    public static void main(String[] args) {
        Map<Integer, List<Integer>> dependencies = new HashMap<>();
        dependencies.put(0, new ArrayList<>());
        dependencies.put(1, Arrays.asList(0));
        dependencies.put(2, Arrays.asList(1));
        dependencies.put(3, Arrays.asList(1, 2));
        dependencies.put(4, Arrays.asList(3));
        System.out.println(new PackageInstaller().install(dependencies));
    }



}
